src/middleware/categoriesMiddleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Request, Response, NextFunction } from 'express';
+import { isArrayofStrings } from '../utils/isArray';
+
+
+
+export const categoriesMiddleware = (req: Request, res: Response, next: NextFunction) => {
```

Comment:
explore open source library for request validation for checking data type and enums allowed